libsolidity/interface/CompilerStack.cpp

```diff
@@ -986,7 +986,12 @@ Json::Value CompilerStack::assemblyJSON(std::string const& _contractName) const
 
 	Contract const& currentContract = contract(_contractName);
 	if (currentContract.evmAssembly)
-		return currentContract.evmAssembly->assemblyJSON(sourceIndices());
+	{
+		std::vector<std::string> sources = sourceNames();
+		if (find(sources.begin(), sources.end(), CompilerContext::yulUtilityFileName()) == sources.end())
+			sources.emplace_back(CompilerContext::yulUtilityFileName());
+		return currentContract.evmAssembly->assemblyJSON(sources);
+	}
```

Comment:
Right. But should the index still to be added in the assembly JSON? As done here: https://github.com/ethereum/solidity/pull/13673/files#diff-786911c6e3eab4698a9fab73ced71946ad895b439784b26b05a0321c2ae0e9ceR503 

Here is an example of the output: https://github.com/ethereum/solidity/pull/13673/files#diff-b490122231e39520a628afc7fd70ef28a17a24eeaec2c75e025c3d04bfb840c9R201

I'm assuming you also meant to revert the change in [Assembly::assemblyJSON](https://github.com/ethereum/solidity/pull/13673/files#diff-786911c6e3eab4698a9fab73ced71946ad895b439784b26b05a0321c2ae0e9ceR435), right?

Reply:
Ops, I'm sorry. I updated the comment. It is related in the sense that `Assembly::assemblyJSON` was changed to work with a vector of source strings instead of a map of source indices (see: https://github.com/ethereum/solidity/blob/aac06c5fa83c5a9aeb768588449337c404ecce22/libevmasm/Assembly.cpp#L435). But I changed it back to work with indices and refactored [EVMAssemblyStack::assemblyJSON](https://github.com/ethereum/solidity/blob/aac06c5fa83c5a9aeb768588449337c404ecce22/libevmasm/EVMAssemblyStack.cpp#L99) to reflect that as well.

Reply:
> First and third link no longer work so not sure which changes you mean, but the `"index"` field seems completely redundant to me. The same value is used as the key in the `".data"`. I wonder why it was added.

It was being used here: https://github.com/ethereum/solidity/pull/13673/commits/8b23b4b78fbf952e2afc13fc1cd4c110eaaf8d46#diff-786911c6e3eab4698a9fab73ced71946ad895b439784b26b05a0321c2ae0e9ceR586 and as you said, it seems redundant, so I removed it and used the value of the key in the `.data` object instead.